Sources/Features/Client/Model/AttachmentsStorage.swift

```
@@ -0,0 +1,129 @@
+        }
+        return attachmentsArray
+    }
+    
+    static func remove(fileName: String) throws {
+        let config = try Config(fileName: fileName)
+        // check file exists
+        guard FileManager.default.fileExists(atPath: config.fileUrl.path) else {
+            throw FileError.fileNotExists
+        }
+        // remove file
+        try FileManager.default.removeItem(at: config.fileUrl)
+        BacktraceLogger.debug("Removed attachments plist at path: \(config.fileUrl)")
+    }
+    
+    private static func convertAttachmentUrlsToBookmarks(_ attachments: Attachments) throws -> Bookmarks {
+        var attachmentsBookmarksDict = Bookmarks()
+        for attachment in attachments {
+            do { let bookmark = try attachment.value.bookmarkData(options: URL.BookmarkCreationOptions.minimalBookmark)
+                attachmentsBookmarksDict[attachment.key] = bookmark as NSData
+            } catch {
+                BacktraceLogger.error("Could not bookmark attachment file URL. Error: \(error)")
+                continue
+            }
+        }
+        return attachmentsBookmarksDict
+    }
+    
+    private static func extractAttachmentUrls(_ bookmarks: Bookmarks) throws -> Attachments {
+        var attachments = Attachments()
+        for bookmark in bookmarks {
+            var stale = ObjCBool(false)
+            guard let fileUrl = try? NSURL.init(resolvingBookmarkData: bookmark.value as Data,
+                                                options: NSURL.BookmarkResolutionOptions(),
+                                                relativeTo: nil,
+                                                bookmarkDataIsStale: &stale) else {
+                BacktraceLogger.error("Could not resolve file URL from bookmark")
+                continue
+            }
+            if stale.boolValue {
+                BacktraceLogger.error("Bookmark data is stale. This should not happen")
```

The error seems to be silently ignored. Is it intentional?

---

We should handle this error since we're not supposed to use stale URLs. Thank you for pointing this out!